main.cpp

```diff
@@ -19,13 +18,31 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+	CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) { }
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+	virtual void run(const MatchFinder::MatchResult& Result) {
+	const auto *CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+	auto &SM = *Result.SourceManager;
+
+	auto DestTypeString = Lexer::getSourceText(CharSourceRange::getTokenRange(
+                               CastExpr->getLParenLoc().getLocWithOffset(1),
+                               CastExpr->getRParenLoc().getLocWithOffset(-1)),
+                               SM, Result.Context->getLangOpts());
+
+	auto s = ("static_cast<" + DestTypeString + ">(").str();
+	auto Range = CharSourceRange::getCharRange(
+			CastExpr->getLParenLoc(),
+			CastExpr->getSubExprAsWritten()->getBeginLoc());
+
+	rewriter_.ReplaceText(Range, s);
+
+	const auto *SubExpr = CastExpr->getSubExprAsWritten()->IgnoreImpCasts();

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.
What would be the difference with the use of `getSubExprAsWritten()->IgnoreImpCasts()`, `getSubExpr()->IgnoreImpCasts()` and `getSubExprAsWritten()`?

Reply:
I use CastExpr->getSubExprAsWritten()->IgnoreImpCasts(), but since getSubExprAsWritten()  retrieve the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes,  I add IgnoreImpCasts().


getSubExpr() retrieves the first cast subexpression, ignoreImpCasts() skips past any implicit casts which might surround this expression until reaching a fixed point. It skips the following AST nodes: ImplicitCastExpr, FullExpr.
I can used get getSubExpr()->IgnoreImpCasts() instead getSubExprAsWritten()->IgnoreImpCasts()

Reply:
`I use CastExpr->getSubExprAsWritten()->IgnoreImpCasts(), but since getSubExprAsWritten() retrieve the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes, I add IgnoreImpCasts().`
Sorry, but I don't understand the reason for using `IgnoreImpCasts()` with `getSubExprAsWritten()` from your explanation.

Reply:
getSubExprAsWritten() retrieves the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes introduced by semantic analysis. getSubExpr() retrieves the first cast subexpression. ignoreImpCasts() skips past any implicit casts which might surround this expression until reaching a fixed point. It skips the following AST nodes: ImplicitCastExpr, FullExpr.

Reply:
Where is the reason? This is just a description of the options above.

Reply:
if we used getSubExprAsWritten() alone or with IgnoreImpCasts(). getSubExprAsWritten()->IgnoreImpCasts(), getSubExpr()->IgnoreImpCasts() and getSubExprAsWritten() give the same result.

Example
```
llvm::outs() << "getSubExprAsWritten()->IgnoreImpCasts(): " << castNode->getSubExprAsWritten()->IgnoreImpCasts()->getStmtClassName();
llvm::outs() << "getSubExpr()->ignoreImpCasts(): " << castNode->getSubExpr()->IgnoreImpCasts()->getStmtClassName();
llvm::outs() << "getSubExprAsWritten(): " << castNode->getSubExprAsWritten()->getStmtClassName();
```
Output:
getSubExprAsWritten()->IgnoreImpCasts(): DeclRefExpr
getSubExpr()->ignoreImpCasts(): DeclRefExpr
getSubExprAsWritten(): DeclRefExpr